dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/data/file/ImmediateFileWriterTest.kt

```diff
@@ -168,4 +173,57 @@ internal class ImmediateFileWriterTest {
         // then
         verifyZeroInteractions(mockDeferredHandler)
     }
+
+    @Test
+    fun `if the file is locked from same JVM(not our case) will handle correctly the exception`(
+        forge: Forge
+    ) {
+        val models = forge.aList { anAlphabeticalString() }
+        val fileNameToWriteTo = forge.anAlphaNumericalString()
+        val file = File(rootDir, fileNameToWriteTo)
+        file.createNewFile()
+        whenever(mockedOrchestrator.getWritableFile(any())).thenReturn(file)
```

Comment:
The `FileOchestrator` provides file through `getWritableFile(itemSize: Int): File` and uses the `itemSize` to consider if `fileHasRoomForMore`. If `fileHasRoomForMore == false`, new file is created.

```kotlin
val fileHasRoomForMore = newSize < maxBatchSize
```

If two processes obtain the same file which has almost the maximum size and write to it, the `maxBatchSize ` will be exceeded. Will this result with data being dropped by Intake? Or the `maxBatchSize ` is only a payload optimisation on the SDKs side?

The same applies to `maxLogPerBatch`:
```kotlin
val fileHasSlotForMore = (lastKnownFileCount < maxLogPerBatch)
```

Reply:
I know that this scenario might happen but I went on this assumption that the batch will not be dropped by the intake in this case 🤔 . Otherwise I should apply the lock also when the file is obtained and I really don't want to do this. @xgouchet do you know if the intake will drop our batch in this case ?